src/components/SQForm/SQForm.tsx

```diff
@@ -1,25 +1,55 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import Grid from '@material-ui/core/Grid';
 import {Formik, Form} from 'formik';
 import {useDebouncedCallback} from 'use-debounce';
 import {object as YupObject, setLocale} from 'yup';
 import {useInitialRequiredErrors} from '../../hooks/useInitialRequiredErrors';
+import type {GridProps} from '@material-ui/core';
+import type {FormikHelpers} from 'formik';
+import type {ObjectShape} from 'yup/lib/object';
 
 setLocale({
   mixed: {
     required: 'Required',
   },
 });
 
-function SQForm({
+interface SQFormProps<Values extends Record<string, unknown>> {
+  /** Form Input(s) */
+  children: React.ReactNode;
+  /** Bool to pass through to Formik. https://formik.org/docs/api/formik#enablereinitialize-boolean */
+  enableReinitialize?: boolean;
+  /** Form Entity Object */
+  initialValues: Values;
+  /** Any prop from https://material-ui.com/api/grid */
+  muiGridProps?: GridProps;
+  /**
+   * Form Submission Handler | @typedef onSubmit: (values: Values, formikBag: FormikBag) => void | Promise<any>
+   * IMPORTANT: If onSubmit is async, then Formik will automatically set isSubmitting to false on your behalf once it has resolved.
+   * This means you do NOT need to call formikBag.setSubmitting(false) manually.
+   * However, if your onSubmit function is synchronous, then you need to call setSubmitting(false) on your own.
+   *
+   * https://jaredpalmer.com/formik/docs/api/withFormik#handlesubmit-values-values-formikbag-formikbag--void--promiseany
+   * */
+  onSubmit?: (
```

Comment:
I know my JS Doc comment above specifies the type. Instead of manually typing `onSubmit` can we use the one from Formik's types?

Reply:
@SeanGroff Are you saying to update the JS Doc comment?  From Fromik, the type for `onSubmit` is `(values: Values, formikHelpers: FormikHelpers<Values>) => void | Promise<any>` which is exactly what I am using except my return type is `void | Promise<unknown>` to avoid eslint yelling at my for an unexpected any.

Reply:
@aharpt Sorry for not being clear. So when I wrote that JS Doc comment I stole the Type from their TypeScript definition.
Does Formik expose an `onSubmit` type that we could use instead of mirror their return type?

Example:

```typescript
`onSubmit: FormikOnSubmit` // or whatever the name of their type is for their onSubmit
```

Reply:
I think
```
import type {FormikConfig} form 'formik'
...
onSubmit: FormikConfig<Values>['onSubmit']
```

Would be the better option that way we can get our generics in there and ensure we have type safety on our `onSubmit` params

Reply:
Will try that.  Thanks @20BBrown14 and @SeanGroff.

Reply:
Looks great, thank you @aharpt